test/apps/iodemo/io_demo.cc

```diff
@@ -217,6 +227,54 @@ class P2pDemoCommon : public UcxContext {
         return conn->send_io_message(m->buffer(), opts().iomsg_size, m);
     }
 
+    bool send_recv_data_as_chunks(UcxConnection* conn, size_t data_size, uint32_t sn,
+                send_recv_data_t send_recv_data, UcxCallback* callback = EmptyCallback::get()) {
+        size_t left_size = data_size;
+        while (left_size > 0) {
+            size_t sz = (left_size >= opts().chunk_size) ? opts().chunk_size : left_size;
+            if (send_recv_data == SEND_DATA) {
```

Comment:
this if can be removed if pass `send_data`/`recv_data` as a function pointer to this function

Reply:
@dmitrygx Since send_data/recv_data are member functions of a class, Passing them as the paremeter of function need  to add some other logic. This is even more complex than the if checking as in above code. So maybe it's not neccessary to change to function pointer here. If any more comments please feel free to let me know. Thanks!